### PR TITLE
Microbenchmark for inter-node object transfer

### PR DESCRIPTION
**Pull request by danyangz**
This pull request adds a microbenchmark to "ray microbenchmark" for object transfer on an emulated multi-node ray cluster.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Comment by pcmoritz** [comment]
There is still some things we need to do before we can merge this:

- Make sure the new cluster setup doesn't interfere with the rest of the tests (maybe we can start the second node after we have run all the other tests, before we are running the multi node test)
- Figuring out the performance difference between the test and starting ray manually

**Comment by pcmoritz** [comment]
These are the raylet parameters for the test:
```
pcmoritz         40076   0.5 11.6  6426888 975124   ??  Ss    6:11PM   0:02.78 /Users/pcmoritz/ray/python/ray/core/src/ray/raylet/raylet --raylet_socket_name=/tmp/ray/session_2019-11-05_18-11-53_715325_40061/sockets/raylet --store_socket_name=/tmp/ray/session_2019-11-05_18-11-53_715325_40061/sockets/plasma_store --object_manager_port=0 --node_manager_port=0 --node_ip_address=192.168.1.5 --redis_address=192.168.1.5 --redis_port=10814 --num_initial_workers=1 --maximum_startup_concurrency=1 --static_resource_list=node:192.168.1.5,1.0,CPU,1,memory,25,object_store_memory,28 --config_list= --python_worker_command=/Users/pcmoritz/anaconda3/bin/python /Users/pcmoritz/ray/python/ray/workers/default_worker.py --node-ip-address=192.168.1.5 --object-store-name=/tmp/ray/session_2019-11-05_18-11-53_715325_40061/sockets/plasma_store --raylet-name=/tmp/ray/session_2019-11-05_18-11-53_715325_40061/sockets/raylet --redis-address=192.168.1.5:10814 --temp-dir=/tmp/ray --java_worker_command= --redis_password= --temp_dir=/tmp/ray --session_dir=/tmp/ray/session_2019-11-05_18-11-53_715325_40061
pcmoritz         40105   0.3 13.0  6474216 1093044   ??  Ss    6:11PM   0:03.85 /Users/pcmoritz/ray/python/ray/core/src/ray/raylet/raylet --raylet_socket_name=/tmp/ray/session_2019-11-05_18-11-53_715325_40061/sockets/raylet.1 --store_socket_name=/tmp/ray/session_2019-11-05_18-11-53_715325_40061/sockets/plasma_store.1 --object_manager_port=0 --node_manager_port=0 --node_ip_address=192.168.1.5 --redis_address=192.168.1.5 --redis_port=10814 --num_initial_workers=1 --maximum_startup_concurrency=1 --static_resource_list=node:192.168.1.5,1.0,CPU,1,GPU,1,memory,30,object_store_memory,28 --config_list= --python_worker_command=/Users/pcmoritz/anaconda3/bin/python /Users/pcmoritz/ray/python/ray/workers/default_worker.py --node-ip-address=192.168.1.5 --object-store-name=/tmp/ray/session_2019-11-05_18-11-53_715325_40061/sockets/plasma_store.1 --raylet-name=/tmp/ray/session_2019-11-05_18-11-53_715325_40061/sockets/raylet.1 --redis-address=192.168.1.5:10814 --temp-dir=/tmp/ray --java_worker_command= --redis_password= --temp_dir=/tmp/ray --session_dir=/tmp/ray/session_2019-11-05_18-11-53_715325_40061
```

**Comment by danyangz** [comment]
ubuntu     8937  0.2  0.0 3007156 9768 ?        Ssl  02:19   0:00 /home/ubuntu/ray/python/ray/core/src/ray/raylet/raylet --raylet_socket_name=/tmp/ray/session_2019-11-06_02-20-00_137574_8923/sockets/raylet --store_socket_name=/tmp/ray/session_2019-11-06_02-20-00_137574_8923/sockets/plasma_store --object_manager_port=0 --node_manager_port=0 --node_ip_address=172.31.8.149 --redis_address=172.31.8.149 --redis_port=6379 --num_initial_workers=64 --maximum_startup_concurrency=64 --static_resource_list=node:172.31.8.149,1.0,CPU,64,memory,4515,object_store_memory,263 --config_list= --python_worker_command=/home/ubuntu/anaconda3/bin/python /home/ubuntu/ray/python/ray/workers/default_worker.py --node-ip-address=172.31.8.149 --object-store-name=/tmp/ray/session_2019-11-06_02-20-00_137574_8923/sockets/plasma_store --raylet-name=/tmp/ray/session_2019-11-06_02-20-00_137574_8923/sockets/raylet --redis-address=172.31.8.149:6379 --temp-dir=/tmp/ray --java_worker_command= --redis_password= --temp_dir=/tmp/ray --session_dir=/tmp/ray/session_2019-11-06_02-20-00_137574_8923

ubuntu    37534  0.2  0.0 3007152 9892 ?        Ssl  02:20   0:00 /home/ubuntu/ray/python/ray/core/src/ray/raylet/raylet --raylet_socket_name=/tmp/ray/session_2019-11-06_02-20-00_137574_8923/sockets/raylet --store_socket_name=/tmp/ray/session_2019-11-06_02-20-00_137574_8923/sockets/plasma_store --object_manager_port=0 --node_manager_port=0 --node_ip_address=172.31.14.178 --redis_address=172.31.8.149 --redis_port=6379 --num_initial_workers=64 --maximum_startup_concurrency=64 --static_resource_list=custom,1,node:172.31.14.178,1.0,CPU,64,memory,4669,object_store_memory,263 --config_list= --python_worker_command=/home/ubuntu/anaconda3/bin/python /home/ubuntu/ray/python/ray/workers/default_worker.py --node-ip-address=172.31.14.178 --object-store-name=/tmp/ray/session_2019-11-06_02-20-00_137574_8923/sockets/plasma_store --raylet-name=/tmp/ray/session_2019-11-06_02-20-00_137574_8923/sockets/raylet --redis-address=172.31.8.149:6379 --temp-dir=/tmp/ray --java_worker_command= --redis_password= --temp_dir=/tmp/ray --session_dir=/tmp/ray/session_2019-11-06_02-20-00_137574_8923

File: python/ray/ray_perf.py
Comment (ericl) [inline python/ray/ray_perf.py]: This will have a hard to guess impact on the other microbenchmarks. Can you make this a separate file?

File: python/ray/ray_perf.py
Comment (ericl) [inline python/ray/ray_perf.py]: Which metrics do we care about, does this capture all of them?

**Review by pcmoritz — APPROVED** [review]
LGTM, can be merged after tests pass.

Additional benchmarks we can add in another PR would be for one to many, many to one transfers or many to many transfers.

**Pull request merged**
